kyuubi-server/src/main/scala/org/apache/kyuubi/operation/ExecuteStatement.scala

```diff
@@ -164,21 +166,18 @@ class ExecuteStatement(
   }
 
   override protected def runInternal(): Unit = {
-    if (shouldRunAsync) {
-      executeStatement()
-      val sessionManager = session.sessionManager
-      val asyncOperation = new Runnable {
-        override def run(): Unit = waitStatementComplete()
-      }
-      try {
-        val backgroundOperation =
-          sessionManager.submitBackgroundOperation(asyncOperation)
-        setBackgroundHandle(backgroundOperation)
-      } catch onError("submitting query in background, query rejected")
-    } else {
-      setState(OperationState.RUNNING)
-      executeStatement()
-      setState(OperationState.FINISHED)
+    executeStatement()
+    val sessionManager = session.sessionManager
+    val asyncOperation: Runnable = () => waitStatementComplete()
+    try {
+      val opHandle = sessionManager.submitBackgroundOperation(asyncOperation)
+      setBackgroundHandle(opHandle)
+    } catch onError("submitting query in background, query rejected")
+
+    if (!shouldRunAsync) try {
+      getBackgroundHandle.get()
+    } catch {
+      case _: ExecutionException => // ignore since state and exception are already updated
```

Comment:
in sync mode, clients do not get operation failure through GetOperationStatus API, but fail with execution failure directly

Reply:
Make sense, updated.